app/ui/src/Pages/Project/pages/KG/components/Filters/components/TopicFilter/TopicFilter.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './TopicFilter.module.scss';
+import { Button, BUTTON_ALIGN, TextInput, useClickOutside } from 'kwc';
+import cx from 'classnames';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import AnimateHeight from 'react-animate-height';
+import TopicList, { TopicListProps } from './components/TopicList/TopicList';
+import SearchIcon from '@material-ui/icons/Search';
+
+interface Props extends TopicListProps {
+  onResetClick: () => void;
+  onFilterChange: (text: string) => void;
+}
+
+function TopicFilter({
+  topics,
+  selectedTopics,
+  onSelectOption,
+  onFilterChange,
+  onResetClick,
+}: Props) {
+  const contentRef = useRef(null);
+  const [opened, setOpened] = useState(false);
+
+  const { addClickOutsideEvents, removeClickOutsideEvents } = useClickOutside({
+    componentRef: contentRef,
+    action: () => setOpened(false),
+  });
+
+  useEffect(() => {
+    if (contentRef && opened) addClickOutsideEvents();
+    else removeClickOutsideEvents();
+    return () => removeClickOutsideEvents();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [contentRef, opened]);
+
+  function toggleFilter() {
+    setOpened(!opened);
+  }
+
+  function getButtonLabel() {
+    const selectedTopicsCount = selectedTopics.length;
+    return `${selectedTopicsCount} TOPIC${
+      selectedTopicsCount !== 1 ? 'S' : ''
+    }`;
```

Comment:
I recently discovered that it is more clean to do this way:
```suggestion
    return `${selectedTopicsCount} ${
      selectedTopicsCount === 1 ? 'TOPIC' : 'TOPICS'
    }`;
```